src/main/kotlin/dsx/bps/crypto/grm/GrmExplorer.kt

```diff
@@ -35,26 +35,28 @@ class GrmExplorer(
 
         timer(this::class.toString(), true, 0, frequency) {
             val newTxId = coin.getLastInternalTxId()
-            val notProcessedGrmTxs = coin.getAccountTxs(newTxId, lastTxId)
+            val notProcessedGrmTxs = coin.getAccountTxs(coin.accountAddress, newTxId, lastTxId)
 
-            for (grmTx in notProcessedGrmTxs.reversedArray()) {
-                processNewGrmTx(grmTx)
+            for (i in notProcessedGrmTxs.size - 1 downTo 0) {
+                processNewGrmTx(notProcessedGrmTxs[i])
             }
             lastTxId = newTxId
         }
     }
 
     private fun processNewGrmTx(grmTx: GrmRawTransaction) {
-        val tx = coin.constructDepositTx(grmTx)
-        if (tx.amount() > BigDecimal.ZERO &&
-            (tx.paymentReference()?.length ?: -1) < 500
-        ) {
-            val newTx = txService.add(
-                tx.status(), tx.destination(), tx.paymentReference(), tx.amount(),
-                tx.fee(), tx.hash(), tx.index(), tx.currency()
-            )
-            grmService.add(grmTx.utime, grmTx.storageFee, grmTx.transactionId.lt, newTx)
-            emitter.onNext(tx)
+        if (grmTx.inMsg.source != "") {
```

Comment:
наноминор-предложение
```suggestion
        if (grmTx.inMsg.source.IsEmpty()) return
```

Reply:
Окей, поправил)